<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RectangleWindowClass</class>
 <widget class="QMainWindow" name="RectangleWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1218</width>
    <height>880</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RectangleWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>591</width>
      <height>751</height>
     </rect>
    </property>
    <property name="title">
     <string>显示窗口</string>
    </property>
    <widget class="CMyLabel" name="lbl_Window">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>571</width>
       <height>501</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtn_ReadImage">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>打开图片</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtn_FitImage">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>自适应</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_Status">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>550</y>
       <width>561</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>坐标灰度信息</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ClearWindow">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>清空窗口</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="ckb_ShowObj">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>600</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>缩放时显示图形</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_Hide">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>650</y>
       <width>111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="text">
      <string>关闭</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget_9">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>30</y>
      <width>581</width>
      <height>731</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_36">
     <attribute name="title">
      <string>卡尺设置</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_172">
            <property name="text">
             <string>中心Row</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_CenterRow_Rectangle">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>999999999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_173">
            <property name="text">
             <string>中心Col</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_CenterCol_Rectangle">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>999999999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_174">
            <property name="text">
             <string>角度</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_Angle_Rectangle">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>-99999.000000000000000</double>
            </property>
            <property name="maximum">
             <double>99999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_175">
            <property name="text">
             <string>半长</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_Len1_Rectangle">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>999999999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="label_176">
            <property name="text">
             <string>半宽</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_Len2_Rectangle">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>999999999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="label_157">
            <property name="text">
             <string>卡尺数量</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spb_MeasureNum_Rectangle">
            <property name="maximum">
             <number>999999</number>
            </property>
            <property name="value">
             <number>40</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="label_158">
            <property name="text">
             <string>搜索长度</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spb_MeasureLength1_Rectangle">
            <property name="maximum">
             <number>999999</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QLabel" name="label_159">
            <property name="text">
             <string>投影长度</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spb_MeasureLength2_Rectangle">
            <property name="maximum">
             <number>999999</number>
            </property>
            <property name="value">
             <number>2</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QLabel" name="label_160">
            <property name="text">
             <string>平滑(Sigma)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_MeasureSigma_Rectangle">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>0.400000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <widget class="QLabel" name="label_161">
            <property name="text">
             <string>阈值</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spb_MeasureThreshold_Rectangle">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_12">
          <item>
           <widget class="QLabel" name="label_162">
            <property name="text">
             <string>极性</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cmb_MeasureTransition_Rectangle">
            <item>
             <property name="text">
              <string>positive</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>negative</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>all</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uniform</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_13">
          <item>
           <widget class="QLabel" name="label_163">
            <property name="text">
             <string>边缘选取</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cmb_MeasureSelect_Rectangle">
            <item>
             <property name="text">
              <string>first</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>last</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>all</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <widget class="QLabel" name="label_164">
            <property name="text">
             <string>最小得分</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_MinScore_Rectangle">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.550000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_15">
          <item>
           <widget class="QLabel" name="label_165">
            <property name="text">
             <string>距离剔除</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dspb_DistanceThreshold_Rectangle">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="maximum">
             <double>100.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>3.500000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_16">
          <item>
           <widget class="QLabel" name="label_166">
            <property name="text">
             <string>插值算法</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cmb_MeasureInterpolation_Rectangle">
            <item>
             <property name="text">
              <string>nearest_neighbor</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>bilinear</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>bicubic</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_17">
          <item>
           <widget class="QLabel" name="label_167">
            <property name="text">
             <string>是否跟随</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cmb_IsFollow_Rectangle">
            <item>
             <property name="text">
              <string>No</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Yes</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <spacer name="verticalSpacer_10">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>204</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_132">
        <item>
         <widget class="QPushButton" name="btn_DrawRectangle_Rectangle">
          <property name="text">
           <string>绘制矩形</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_DetectRectangle_Rectangle">
          <property name="text">
           <string>检测矩形</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_ClearRoi_Rectangle">
          <property name="text">
           <string>清除ROI</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="tab_37">
     <attribute name="title">
      <string>结果</string>
     </attribute>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>12</x>
        <y>28</y>
        <width>541</width>
        <height>651</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_133">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_134">
           <item>
            <widget class="QPushButton" name="btn_SaveData_Rectangle">
             <property name="text">
              <string>参数保存</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_LoadData_Rectangle">
             <property name="text">
              <string>参数加载</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="lbl_RunCT_Rectangle">
           <property name="text">
            <string>运行时间</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_11">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>428</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="label_168">
             <property name="text">
              <string>中点X:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_169">
             <property name="text">
              <string>中点Y:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_170">
             <property name="text">
              <string>旋转角度:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_171">
             <property name="text">
              <string>半长:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_187">
             <property name="text">
              <string>半宽:</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLineEdit" name="txt_Col_Rectangle">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="txt_Row_Rectangle">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="txt_Angle_Rectangle">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="txt_Len1_Rectangle">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="txt_Len2_Rectangle">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_38">
     <attribute name="title">
      <string>点结果</string>
     </attribute>
     <widget class="QTableWidget" name="tablewidget_PointsResult_Rectangle">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>561</width>
        <height>671</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1218</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CMyLabel</class>
   <extends>QLabel</extends>
   <header>cmylabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
